changelogs/3.3.2.md

```diff
@@ -0,0 +1,193 @@
+type in them**. What is worse, just an alter `double` -> `number` **won't
+work**. The old index must be dropped and a new one must be created. In any
+order. See also gh-9965 and gh-11017 for more details.
+
+Please, consider the full list of user-visible changes below.
+
+## Compatibility
+
+Tarantool 2.x and 3.x are compatible in the binary data layout, client-server
+protocol, and replication protocol. It means that the updating may be performed
+with zero downtime for read requests and the-order-of-network-lag downtime for
+write requests.
+
+Please, follow the [upgrade procedure][upgrade] to plan your update actions.
+
+Users of Tarantool 2.x may be interested in the [compat][compat] options that
+allow to imitate some 2.x behavior. This allows to perform application code
+update step-by-step, not all-at-once.
+
+[compat]: https://www.tarantool.io/en/doc/latest/reference/configuration/configuration_reference/#compat
+
+[upgrade]: https://www.tarantool.io/en/doc/latest/book/admin/upgrades/
+
+## Functionality added or changed
+
+### Tools
+
+Made LuaJIT profilers more user-friendly:
+* `misc.memprof.start()` without arguments writes the dump into the default file
+  named `memprof.bin` instead of raising an error.
+* `misc.sysprof.start()` provides more verbose errors in case of profiler
+  misuse.
+* If the profiler is disabled for the target platform, it is now mentioned in
+  the error message explicitly.
+* `misc.sysprof.start()` without arguments starts the profiler in the default
+  mode `"D"`.
+
+## Bugs fixed
+
+### Core
+
+* Fixed a bug when the `sequence:reset()` call result was not recovered
+  after the server restart (gh-9871).
+* Fixed log message being truncated to 1024 bytes with JSON logger (gh-10918).
+* Fixed several bugs in the numbers comparison and hashing in tuple keys. It
+  allowed to insert the same key multiple times into a unique index, and
+  sometimes wouldn't allow to find an existing key in an index. Could happen
+  when numbers were encoded in MessagePack suboptimally and when the `double`
+  field type was used (gh-9965).
+* Fixed the issue with hanging write operations forever triggered by heavy
+  write load and WAL writing failures on cascade rollback (gh-11081).
+* Fixed a bug when cancelling a fiber waiting in WAL queue corrupted the
+  the WAL (gh-11078).
+* Fixed a bug when the timestamps of snapshots created before the server restart
+  were not taken into account with `checkpoint_interval` enabled (gh-9820).
+* Fixed a bug when the sizes of xlog files created before the server restart
+  were not taken into account during the `checkpoint_wal_threshold` exceedance
+  checks (gh-9811).
+* Fixed a bug when transactions in the WAL queue were not rolled back on WAL
+  error (gh-11179).
+* Fixed a bug when WAL queue was not flushed properly. In particular
+  when building index of vinyl space. In the latter case it may lead
+  the new index missing data from transactions in the queue
```

Comment:
```suggestion
* Fixed a bug when the WAL queue was not flushed properly. In particular
  when building an index of a vinyl space. In the latter case it might make
  the new index miss data from transactions in the queue
```